### ISSUE TITLE
[OAuth 2][Multiaccount] Determine how to login into multiple logged out accounts

### ISSUE THREAD

**Issue by SamuAlfageme** [owncloud/client#5895]
### Steps to reproduce

1 . Login to one OAuth server using the client
2. Open server in the web browser & log out
3. Repeat 1 **with a different account** on the client
4. One of the following:
 - Log out of both accounts on the client
 - Remove both authorization & refresh tokens from the server DB 
 - Delete the cookies & keychain entries from the client
5. Restart the client

Just one login page is triggered from the client, the rest of the accounts will stay in the same state as the one in https://github.com/owncloud/client/issues/5893 ("Configuration Error") until all the accounts complete the log-in process from the account tab / the client is restarted again as many times as accounts logged-out. 

### Actual behavior
Depending on which account is the one launching the auth. flow from the client, just one (the first) is able to login. If using the wrong credentials a message is displayed:

<p align="center">
  <img src="https://user-images.githubusercontent.com/2644445/28127430-d2c75ba6-672c-11e7-844c-4ee063953b55.png"/>
</p>

...but up until this point, the user does not know on which account is he trying to login to. Also, since the session is kept in the browser after each authorization, it's quite tricky to recover and re-login to all accounts.

### Expected behavior
Don't really know how the best way to deal with this scenario. Might as well do some research on how other multi-account applications that use OAuth2 w/ the browser do it. 

Some ideas: 

* In SAML auth flow we spawned as many independent web-views as accounts created in the client
    - Maybe we could force the browser to launch in "incognito mode" (e.g. flag to `xdg-open`) - but I don't really think this is even possible / an optimal cross-platform solution.
* Queue the logins combined with https://github.com/owncloud/oauth2/issues/48
    - cc/ @ogoffart should we escalate this to the oauth2 repo to force the application logins/authorizations not to leave a valid session open behind on the browser? 
* **Do not trigger** any login attempt and leave the user do it manually. 

cc/ @DeepDiver1975 @butonic

**Comment by ogoffart** [comment]
What could be done better already in the client is that we could show the error in the browser instead of redirecting to the login page.

I believe otherwise the best way is that we tell the oauth app a hint about the user, as i wrote in owncloud/oauth2#48  then there would be different tab/window for each, and the user will have to authenticate on each tab.  

>  Maybe we could force the browser to launch in "incognito mode" 

One of the advantage of the oauth process is that it does not require entering the password if one is logged in the browser already. So doing that would be unfortunate.
Also that's not easy to do while supporting all platform and browsers.

**Comment by SamuAlfageme** [comment]
@ogoffart

> I believe otherwise the best way is that we tell the oauth app a hint about the user, as i wrote in owncloud/oauth2#48 then there would be different tab/window for each

Agreed, owncloud/oauth2#48 could be a starting approach. Maybe also passing a query.param. to say not to preserve session after each login can be implemented in the OAuth app that will allow us to open as many "pre-filled" login forms on the browser and address them independently. 

> One of the advantage of the oauth process is that it does not require entering the password if one is logged in the browser already. So doing that would be unfortunate. 

But I believe some apps enforce reentering those / even create a complete disposable webview for this purpose. I understand the huge advantage of auto-login but might carry side effects.

**Comment by ogoffart** [comment]
Pull request to put the message in the browser instead of a message box. https://github.com/owncloud/client/pull/5901

**Comment by SamuAlfageme** [comment]
🔐 - Blocked by possible regression in https://github.com/owncloud/oauth2/issues/66

**Comment by ckamm** [comment]
I've just tested this. 

I logged out of two oauth2 accounts on the same server and restarted the client. Two login windows popped up in the browser to log me in again. (one needed me to log into the right account first) Using these worked.

**Comment by SamuAlfageme** [comment]
We recently got some negative feedback on how complicated it is to do multiple, simultaneous re-logins on the client. Users get overwhelmed and confused with many (non-reactive to each other) tabs. 

By the time the user has logged in in one tab and closes it, next one still prompts a login form. User is set, but browser session belongs to a different user: "switch accounts" view will be displayed and the original log in form will follow. 

I'm thinking on rescuing the idea from https://github.com/owncloud/client/issues/5895#issue-242438883

> Queue the logins combined with owncloud/oauth2#48

Re-login shouldn't ideally happen that often tho. Something to consider.

**Comment by guruz** [comment]
> There are also ways to prefill the user

Is that URL generated in the oauth2 app or in the client?
CC @ogoffart @DeepDiver1975

**Comment by michaelstingl** [comment]
As far as I know, the `index.php/login?user=iamtheuser` isn't used in context of OAuth2 yet.

**Comment by ogoffart** [comment]
@guruz 
> Is that URL generated in the oauth2 app or in the client?

The client generated URL is tot the 'index.php/apps/oauth2/authorize` endpoints, and it has the `user=...`  query parameter.  Then the server redirects to the login page, using the original url from the client as a redirect url. The javascript from the login page is aware of that and use the user= from the authorize url to pre-fill the username


@SamuAlfageme 

So your suggestion would be to queue the opening of browser. So wait that one account is connected until we open a browser for the other account?
But this may be a problem for users who don't wish to login to one account, but wants to login with another account (although they could go in the settings dialog and click the link to open the page)
Please open another issue for queueing the browser opening.

**Comment by SamuAlfageme** [comment]
@ogoffart I haven't thought about this enough, not sure if plain "enqueuing" the logins is the best solution; since I fully agree with your concerns on: 

> But this may be a problem for users who don't wish to login to one account, but wants to login with another account (although they could go in the settings dialog and click the link to open the page)

Maybe disable auto-opening the browser for the scenario where >1 `http_oauth=true` accounts are in `CredentialsWrong` state? produce a notification and let user do it manually?
